sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/RedundantConditionalAroundAssignment.cs

```diff
@@ -48,7 +49,8 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                 {
                     var ifStatement = (IfStatementSyntax)c.Node;
                     if (ifStatement.Else != null ||
-                        ifStatement.Parent is ElseClauseSyntax)
+                        ifStatement.Parent is ElseClauseSyntax ||
+                        ifStatement.Ancestors().OfType<PropertyDeclarationSyntax>().Any())

```

Comment:
I am not sure if there is any difference but since Roslyn provides a `FirstAncestorOrSelf<T>` it might be better to use it rather than `Ancestors().OfType<PropertyDeclarationSyntax>().Any()`.

Reply:
That is interesting. They boil down to the same thing, but `OfType()` does two casts (is and then C-style cast), which I presume is slower.